src/internal/core.rs

```diff
@@ -248,10 +241,10 @@ impl<DP: DependencyProvider> State<DP> {
 
     /// Add this incompatibility into the set of all incompatibilities.
     ///
-    /// Pub collapses identical dependencies from adjacent package versions
+    /// PubGrub collapses identical dependencies from adjacent package versions
```

Comment:
I only touched this because my `s/pub /pub(crate) /g` misfired here, but this seems more correct?

Reply:
I think `Pub` here was intentional, is it was a reference to the dart implementation and documentation that is called `Pub`

Reply:
I believe this is something we haven’t implemented yet in (rust) pubgrub

Reply:
Then we should be more explicit, "The original PubGrub implementation in Dart ..."

Reply:
This particular optimization has been implemented in https://github.com/pubgrub-rs/pubgrub/pull/163, although only for the special case described in this comment. So at this point any name seems fine.